internal/releaseversion/releaseversion.go

```diff
@@ -0,0 +1,104 @@
+package releaseversion
+
+import (
+	"context"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/giantswarm/apiextensions/v2/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/azure-admission-controller/internal/errors"
+)
+
+const (
+	versionLabel = "release.giantswarm.io/version"
+)
+
+func GetVersionFromCRLabels(labels map[string]string) (semver.Version, error) {
+	version, ok := labels[versionLabel]
+	if !ok {
+		return semver.Version{}, microerror.Maskf(errors.ParsingFailedError, "CR didn't have expected label %s", versionLabel)
+	}
+
+	return GetVersionFromString(version)
+}
+
+func GetVersionFromString(version string) (semver.Version, error) {
+	return semver.ParseTolerant(version)
+}
+
+func IsAlphaRelease(release string) bool {
+	return strings.Contains(release, "alpha")
+}
+
+func UpgradeAllowed(ctx context.Context, g8sclient versioned.Interface, oldVersion semver.Version, newVersion semver.Version) (bool, error) {
+	if !oldVersion.Equals(newVersion) {
```

Comment:
small style comment: I believe it's easier to read returning earlier and one less indentation level
```
if oldVersion.Equals(newVersion) {
    return true, nil
}

availableReleases, err := availableReleases(ctx, g8sclient)
...
..
.
```

Reply:
Agreed, done